### PR TITLE
Add model cards

### PR DESCRIPTION
**Pull request by Pringled**
This PR adds:
- A model card template
- Automatic model card generation and saving when using save_pretrained()
- Support for pushing model cards

### CODE REVIEW

File: model2vec/utils.py [hunk 1]
```diff
@@ -33,14 +35,23 @@ def setup_logging() -> None:
...
 
     :param folder_path: The path to the folder.
-    :param embeddings: The embeddings
-    :param tokenizer: The tokenizer
-    :param config: A metadata config
+    :param embeddings: The embeddings.
+    :param tokenizer: The tokenizer.
+    :param config: A metadata config.
+    :param create_model_card: Whether to create a model card.
+    :param **kwargs: Any additional arguments.
```
Comment (stephantul) [inline model2vec/utils.py:54]: Should note that this goes to `_create_model_card`

File: model2vec/utils.py [hunk 2]
```diff
@@ -49,6 +60,53 @@ def save_pretrained(folder_path: Path, embeddings: np.ndarray, tokenizer: Tokeni
...
+    template_path = pkg_resources.resource_filename("model2vec", "model_card_template.md")
+    with open(template_path, "r") as file:
+        template_content = file.read()
+
+    # # Fill in the placeholders in the template
+    model_card_content = template_content.format(
+        model_name=model_name,
+        base_model=kwargs.get("base_model_name", "unknown"),
+        language=kwargs.get("language", "unknown"),
+        license=kwargs.get("license", "mit"),
```
Comment (stephantul) [inline model2vec/utils.py]: maybe dangerous to default to `mit`

File: model2vec/utils.py [hunk 3]
```diff
@@ -49,6 +60,53 @@ def save_pretrained(folder_path: Path, embeddings: np.ndarray, tokenizer: Tokeni
...
+    :param template_path: The path to the template file.
+    :param **kwargs: Additional metadata for the model card (e.g. language, base_model, etc.).
+    :return: The content of the model card in markdown format.
+    """
+    # Load the template
+    template_path = pkg_resources.resource_filename("model2vec", "model_card_template.md")
+    with open(template_path, "r") as file:
+        template_content = file.read()
+
+    # # Fill in the placeholders in the template
```
Comment (stephantul) [inline model2vec/utils.py]: double `#`

File: model2vec/utils.py [hunk 4]
```diff
@@ -49,6 +60,53 @@ def save_pretrained(folder_path: Path, embeddings: np.ndarray, tokenizer: Tokeni
...
+
+    # Generate the model card content
+    model_card = _generate_model_card(model_name, **kwargs)
+
+    # Save the model card as README.md
+    with open(folder_path / "README.md", "w", encoding="utf8") as fOut:
+        fOut.write(model_card)
+
+
+def _generate_model_card(model_name: str, template_path: str = "assets/model_card_template.md", **kwargs: Any) -> str:
```
Comment (stephantul) [inline model2vec/utils.py:86]: I think `template_path` here doesn't do anything.

File: model2vec/utils.py [hunk 5]
```diff
@@ -49,6 +60,53 @@ def save_pretrained(folder_path: Path, embeddings: np.ndarray, tokenizer: Tokeni
...
+    """
+    # Load the template
+    template_path = pkg_resources.resource_filename("model2vec", "model_card_template.md")
+    with open(template_path, "r") as file:
+        template_content = file.read()
+
+    # # Fill in the placeholders in the template
+    model_card_content = template_content.format(
+        model_name=model_name,
+        base_model=kwargs.get("base_model_name", "unknown"),
```
Comment (stephantul) [inline model2vec/utils.py]: if you expect these arguments, I wouldn't put them in kwargs, but just give them names and these defaults. The behavior is the same, it's just typed better.

File: model2vec/utils.py [hunk 6]
```diff
@@ -49,6 +60,60 @@ def save_pretrained(folder_path: Path, embeddings: np.ndarray, tokenizer: Tokeni
...
+    Create a model card and store it in the specified path.
+
+    :param folder_path: The path where the model card will be stored.
+    :param **kwargs: Additional metadata for the model card (e.g., model_name, base_model, etc.).
+    """
+    folder_path = Path(folder_path)
+    model_name = folder_path.name
+
+    # Generate the model card content
+    model_card = _generate_model_card(model_name, **kwargs)
```
Comment (stephantul) [inline model2vec/utils.py:79]: I wouldn't mind this function containing the function body of `_generate_model_card`. I think the naming is a bit confusing, so maybe just put the body of `_generate_model_card` here?

File: model2vec/utils.py [hunk 7]
```diff
@@ -49,6 +60,60 @@ def save_pretrained(folder_path: Path, embeddings: np.ndarray, tokenizer: Tokeni
...
+
+    # Save the model card as README.md
+    with open(folder_path / "README.md", "w", encoding="utf8") as fOut:
```
Comment (stephantul) [inline model2vec/utils.py:82]: `fOut` is camelCase

Comment (stephantul) [reply model2vec/utils.py:82]: letterlijk `fout`

**Review by stephantul — APPROVED** [review]
Small comments, but LGTM

File: model2vec/utils.py (hunk 4 above)
Comment (Pringled) [reply model2vec/utils.py:86]: You're right, legacy

**Review by stephantul — APPROVED** [review]

**Pull request merged**